### PR TITLE
glew: enable both EGL and GLX

### PR DESCRIPTION
**Pull request by flokli**
Re-roll of https://github.com/NixOS/nixpkgs/pull/328907, but this time adding the patch from ArchLinux, which keeps both EGL and GLX code paths active.

Remove overrides where EGL was explicitly requested previously, as well as the glew-egl package variant.

I ran some games, which seemed to use the GLX path successfully, but this could get some more testing. @K900 can you check if the regressions mentioned in https://github.com/NixOS/nixpkgs/pull/328907#issuecomment-2243834460 are working now?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by K900** [comment]
Can go to master

**Comment by flokli** [comment]
> Can go to master

@K900 did you test the things you mentioned were broken? I didn't white understand what, and how to test it by myself.

**Comment by K900** [comment]
I'm building it now.

**Comment by K900** [comment]
mangoapp works fine with this applied on top of latest unstable-small.

**Comment by flokli** [comment]
Rebased and targeted master, PTAL.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -20401,11 +20398,6 @@ with pkgs;
   glew110 = callPackage ../development/libraries/glew/1.10.nix {
     inherit (darwin.apple_sdk.frameworks) AGL OpenGL;
   };
-  glew-egl = callPackage ../development/libraries/glew {
-    inherit (darwin.apple_sdk.frameworks) OpenGL;
-    enableEGL = true;
-  };
-
```
Comment (SomeoneSerge) [inline pkgs/top-level/all-packages.nix:20408]: I'd start with a `warn` in `aliases.nix`

File: pkgs/top-level/all-packages.nix
Comment (SomeoneSerge) [inline pkgs/top-level/all-packages.nix:1]: It's sad we don't know how to reproduce the failure in a unit test, but I guess it's OK

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (flokli) [reply pkgs/top-level/all-packages.nix:20408]: Done

**Pull request merged**